src/mlx/jira_juggler.py

```diff
@@ -385,29 +474,47 @@ def juggle(self, output=None, **kwargs):
         return juggler_tasks
 
     @staticmethod
-    def link_to_preceding_task(tasks):
+    def link_to_preceding_task(tasks, weeklymax=5.0, current_date=datetime.now()):
         """Links task to preceding task with the same assignee.
 
-        If it's the first task for a given assignee and it's not linked with 'depends on'/'is blocked by' through JIRA,
-        'start ${now}' is added instead.
+        If the task has been resolved, 'end' is added instead no matter what, followed by the date and time on which
+        it's been resolved.
+
+        If it's the first unresolved task for a given assignee and it's not linked with 'depends on'/'is blocked by'
+        through JIRA, 'start' is added instead followed by the date and hour on which the task has been started,
+        i.e. current time minus time spent.
 
         Args:
             tasks (list): List of JugglerTask instances to modify
+            weeklymax (float): Number of allocated workdays per week
+            current_date (datetime.datetime): Offset-naive datetime to treat as the current date
         """
-        assignees_to_tasks = {}
+        unresolved_tasks = {}
         for task in tasks:
             assignee = str(task.properties['allocate'])
-            if assignee in assignees_to_tasks:
-                preceding_task = assignees_to_tasks[assignee][-1]
-                task.properties['depends'].append_value(to_identifier(preceding_task.key))
-                assignees_to_tasks[assignee].append(task)
+            if assignee not in unresolved_tasks:
+                unresolved_tasks[assignee] = []
+
+            depends_property = task.properties['depends']
+            if task.is_resolved:
+                depends_property.PREFIX = ''
+                depends_property.name = 'end'
+                depends_property.value = [task.resolved_at_repr]  # overwrite any links in JIRA
             else:
-                assignees_to_tasks[assignee] = [task]
-                depends_property = task.properties['depends']
-                if not depends_property.value:
-                    depends_property.name = 'start'
+                if unresolved_tasks[assignee]:  # task with dependency
+                    preceding_task = unresolved_tasks[assignee][-1]
+                    depends_property.append_value(to_identifier(preceding_task.key))
+                elif not depends_property.value:  # first unresolved task for assignee
                     depends_property.PREFIX = ''
-                    depends_property.append_value('${now}')
+                    depends_property.name = 'start'
+                    val = to_juggler_date(current_date)
+                    if task.issue.fields.timespent:
+                        days_spent = task.issue.fields.timespent // 3600 / 8
+                        weekends = calculate_weekends(current_date, days_spent, weeklymax)
```

Comment:
twice I was a bit drawn to this. Are we sure we can't get this information somehow from Jira as sprint reports there also take weekends into the account (its flat line over the weekend).

Reply:
The days off are configured in our `preamble.tjp` file like so: `workinghours sat, sun off`. I don't want jira-juggler to parse this file, so I added `--weeklymax` instead and added the assumption that Sunday is the most likely day of the week to be an off-day, followed by Saturday and so on. I've discussed this with @Ben2022 and decided that this approach is good enough (for now).

We'll probably use `--weeklymax 4`  since the `resource` configuration in our `preamble.tjp` contains `limits {weeklymax 32h}` for most developers.

The workingdays and workinghours configured in JIRA are not accessible via their public API as far as I know. Even when we find a way to access this data, is it guaranteed to contain the same configuration as our `preamble.tjp`?